src/Sulu/Component/Symfony/CompilerPass/Customizer/AddArgumentCustomizer.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Symfony\CompilerPass\Customizer;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+class AddArgumentCustomizer implements CustomizerInterface
+{
+    /**
+     * @var mixed
+     */
+    private $argument;
+
+    public function __construct($argument)
+    {
+        $this->argument = $argument;
+    }
+
+    public function customize(Definition $definition, ContainerBuilder $container)
+    {
+        $definition->addArgument($this->argument);
```

Comment:
I don't really see the advantage of this class... It is even more complicated to build a service now, especially if you are used to the tools Symfony already gives you.

Reply:
these classes are used to customize existing service (overwrite and extend it)